### PR TITLE
Storage Supported Providers hyperlink is updated to refer to the correct path

### PR DESCRIPTION
**Pull request by srinivasa-vasu**
Update: Storage Supported Providers hyperlink is updated to refer to the correct path

- [x] [Accepted the DCO](https://velero.io/docs/v1.5/code-standards/#dco-sign-off). Commits without the DCO will delay acceptance.
- [x] [Created a changelog file](https://velero.io/docs/v1.5/code-standards/#adding-a-changelog) or added `/kind changelog-not-required`.

### CODE REVIEW

File: changelogs/unreleased/3667-srinivasa-vasu [hunk 1]
```diff
@@ -0,0 +1 @@
+update storage provider setup link references
```
Comment (jenting) [inline changelogs/unreleased/3667-srinivasa-vasu:1]: It's a doc change, changelog is no needed.

**Review by jenting — CHANGES_REQUESTED** [review]

File: changelogs/unreleased/3667-srinivasa-vasu (hunk 1 above)
Comment (srinivasa-vasu) [reply changelogs/unreleased/3667-srinivasa-vasu]: sure, removed

**Review by jenting — DISMISSED** [review]

File: site/content/docs/main/supported-providers.md [hunk 2]
```diff
@@ -14,8 +14,8 @@ Velero supports a variety of storage providers for different backup and snapshot
...
-| [Container Storage Interface (CSI)](https://github.com/vmware-tanzu/velero-plugin-for-csi/)| 🚫                 | CSI Volumes                  | [Velero plugin for CSI](https://github.com/vmware-tanzu/velero-plugin-for-csi/)             | [CSI Plugin Setup](website-guidelines.md)        |
+| [VMware vSphere](https://github.com/vmware-tanzu/velero-plugin-for-vsphere)              | 🚫                  | vSphere Volumes              | [VMware vSphere](https://github.com/vmware-tanzu/velero-plugin-for-vsphere)                    | [vSphere Plugin Setup](https://github.com/vmware-tanzu/velero-plugin-for-vsphere#velero-plugin-for-vsphere-installation-and-configuration-details)    |
+| [Container Storage Interface (CSI)](https://github.com/vmware-tanzu/velero-plugin-for-csi/)| 🚫                 | CSI Volumes                  | [Velero plugin for CSI](https://github.com/vmware-tanzu/velero-plugin-for-csi/)             | [CSI Plugin Setup](https://github.com/vmware-tanzu/velero-plugin-for-csi#kinds-of-plugins-included)        |
```
Comment (ashish-amarnath) [inline site/content/docs/main/supported-providers.md:18]: Please add the same change to the `site/content/docs/v1.[456]/supported-providers.md` file too.

**Review by ashish-amarnath — CHANGES_REQUESTED** [review]
Thanks for fixing up this documentation.
Can you please make the same change for other versions of the docs too, specifically,
`site/content/docs/v1.4/supported-providers.md`, site/content/docs/v1.5/supported-providers.md`, and site/content/docs/v1.6/supported-providers.md`

**Comment by ashish-amarnath** [comment]
@srinivasa-vasu Are you able to address the comment on the PR?

File: site/content/docs/main/supported-providers.md (hunk 2 above)
Comment (srinivasa-vasu) [reply site/content/docs/main/supported-providers.md:18]: Apologies for the late revert

**Review by ashish-amarnath — APPROVED** [review]
The CSI plugin repo doesn't have specific install instructions.
I am good with getting this merged as-is and readdressing this after we add install instructions in the plugin repo.

File: site/content/docs/main/supported-providers.md (hunk 2 above)
Comment (ashish-amarnath) [reply site/content/docs/main/supported-providers.md:18]: no worries! Thank you for addressing the comment.

**Review by jenting — APPROVED** [review]

**Pull request merged**